imagecraft/pack/grubutil.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""GRUB utils."""
+
+import subprocess
+from pathlib import Path
+
+from craft_cli import emit
+
+from imagecraft import errors
+from imagecraft.pack.chroot import Chroot, Mount
+from imagecraft.pack.image import Image
+from imagecraft.subprocesses import run
+
+_ARCH_TO_GRUB_EFI_TARGET: dict[str, str] = {
+    "amd64": "x86_64-efi",
+    "arm64": "arm64-efi",
+    "armhf": "arm-efi",
+}
+
+
+def _grub_install(grub_target: str, loop_used: str) -> None:
+    grub_install_command = [
+        "grub-install",
+        loop_used,
+        "--boot-directory=/boot",
+        "--efi-directory=/boot/efi",
+        f"--target={grub_target}",
+        "--uefi-secure-boot",
+        "--no-nvram",
+    ]
+
+    update_grub_command = [
+        "update-grub",
+    ]
+
+    os_prober = "/etc/grub.d/30_os-prober"
+    divert_base_command = "dpkg-divert"
+
+    divert_common_args = [
+        "--local",
+        "--divert",
+        os_prober + ".dpkg-divert",
+        "--rename",
+        os_prober,
+    ]
+
+    divert_os_prober_command = [divert_base_command, *list(divert_common_args)]
+
+    undivert_os_prober_command = [
+        divert_base_command,
+        "--remove",
+        *list(divert_common_args),
```

Comment:
It should work without list()
```suggestion
        *divert_common_args,
```